src/benchmarks/micro/libraries/System.Collections/PriorityQueue/Perf_PriorityQueue.cs

```diff
@@ -0,0 +1,142 @@
+    public abstract class Perf_PriorityQueue<TElement, TPriority>
+    {
+        [Params(10, 100, 1000, 10_000, 100_000)]
+        public int Size;
+
+        private (TElement Element, TPriority Priority)[] _items;
+        private PriorityQueue<TElement, TPriority> _priorityQueue;
+        private PriorityQueue<TElement, TPriority> _prePopulatedPriorityQueue;
+
+        public abstract IEnumerable<(TElement Element, TPriority Priority)> GenerateItems(int count);
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _items = GenerateItems(Size).ToArray();
+            _priorityQueue = new PriorityQueue<TElement, TPriority>(initialCapacity: Size);
+            _prePopulatedPriorityQueue = new PriorityQueue<TElement, TPriority>(_items);
+        }
+
+        [Benchmark]
+        public void HeapSort()
+        {
+            var queue = _priorityQueue;
+            queue.EnqueueRange(_items);
+
+            while (queue.Count > 0)
+            {
+                queue.Dequeue();
+            }
+        }
+
+        [Benchmark]
+        public void Enumerate()
+        {
+            foreach (var _ in _prePopulatedPriorityQueue.UnorderedItems)
+            {
+
+            }
+        }
+
+        [Benchmark]
+        public void Dequeue_And_Enqueue()
+        {
+            // benchmarks dequeue and enqueue operations
+            // for heaps of fixed size.
+
+            var queue = _priorityQueue;
+            var items = _items;
+
+            // populate the heap: incorporated in the 
+            // benchmark to achieve determinism.
+            foreach ((TElement element, TPriority priority) in items)
+            {
+                queue.Enqueue(element, priority);
+            }
+
+            foreach ((TElement element, TPriority priority) in items)
+            {
+                queue.Dequeue();
+                queue.Enqueue(element, priority);
+            }
+
+            // Drain the heap
+            while (queue.Count > 0)
+            {
+                queue.Dequeue();
+            }
+        }
+
+        [Benchmark]
+        public void K_Max_Elements()
+        {
+            const int k = 5;
+            var queue = _priorityQueue;
+            var items = _items;
+
+            for (int i = 0; i < k; i++)
+            {
+                (TElement element, TPriority priority) = items[i];
+                queue.Enqueue(element, priority);
+            }
+
+            for (int i = k; i < Size; i++)
+            {
+                (TElement element, TPriority priority) = items[i];
+                queue.EnqueueDequeue(element, priority);
+            }
+
+            for (int i = 0; i < k; i++)
+            {
+                queue.Dequeue();
+            }
+        }
+    }
+
+    public class Perf_PriorityQueue_String_String : Perf_PriorityQueue<string, string>
+    {
+        public override IEnumerable<(string Element, string Priority)> GenerateItems(int count)
```

Comment:
Please use [ValuesGenerator](https://github.com/dotnet/performance/blob/master/src/harness/BenchmarkDotNet.Extensions/ValuesGenerator.cs) instead of generating the test data on your own (it supports both arrays of strings and integers).

https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#always-the-same-input-data